lib/action_subscriber/dsl.rb

```diff
@@ -95,7 +149,12 @@ def _run_action_with_filters(env, action)
       final_block = Proc.new { subscriber_instance.public_send(action) }
 
       first_proc = around_filters.reverse.reduce(final_block) do |block, filter|
-        Proc.new { subscriber_instance.send(filter, &block) }
+        if filter.matches(action)
+          p = Proc.new { subscriber_instance.send(filter.method, &block) } if filter.matches(action)
```

Comment:
I think you can drop the `p =` and `if filter.matches` since you already verified that in this if statement.